scrapers_next/nd/committees.py

```diff
@@ -0,0 +1,90 @@
+from spatula import URL, HtmlListPage, HtmlPage, XPath
+from openstates.models import ScrapeCommittee
+import re
+
+
+class SubcommitteeFound(BaseException):
+    def __init__(self, com_name):
+        super().__init__(
+            f"Scraper has no process for ingesting subcommittee classification: {com_name}"
+        )
+
+
+class CommitteeDetail(HtmlPage):
+    def process_page(self):
+        com = self.input
+        staff = {"staff": []}
+        members = XPath('//div[@class="member-wrapper"]').match(self.root)
+        for each_member in members:
+            member_detail = each_member.text_content().strip()
+            member_detail = [x.strip() for x in member_detail.splitlines() if x.strip()]
+
+            if len(member_detail) == 5:
+                role = member_detail[-1]
+            else:
+                role = "Member"
+            name = [
+                x.text_content()
+                for x in each_member.findall("div", {"class": "strong member-name"})
+            ]
+
+            if len(name) == 1:
+                name = re.sub("\\s+", " ", name[0]).strip()
+                staff["staff"].append({"name": name, "role": "staff"})
+                continue
+            elif len(name) == 3:
+                name = f"{name[0]}{name[1]}"
+                name = re.sub("\\s+", " ", name).strip()
+                staff["staff"].append(
+                    {"name": name, "role": "Committee Citizen Members"}
+                )
+                continue
+            else:
+                name = member_detail[1]
+            com.add_member(name, role)
+        com.extras = staff
+        com.add_source(
+            self.source.url,
+            note="Committee Details page",
+        )
+        return com
+
+
+class CommitteeList(HtmlListPage):
+    source = URL("https://www.ndlegis.gov/assembly/68-2023/committees", timeout=10)
+
+    def process_page(self, only_name=False):
+        all_comm_elements = []
+        for comm_type in ["senate", "house", "interim"]:
+            all_comm_elements += XPath(
+                f"//a[contains(@href, '/committees/{comm_type}/')]"
+            ).match(self.root)
+
+        for elem in all_comm_elements:
+            comm_url = elem.get("href")
```

Comment:
Replace these two lines with the following 14 lines in the code block below. We want to get [statutory committees](https://www.ndlegis.gov/assembly/68-2023/regular/north-dakota-statutory-committees-contain-legislative-members) too. I went ahead and wrote out the code that will do that in this case (it will require imports of `requests` and `lxml.html`):
```
        stat_list_url = XPath(
            "//a[contains(text(), 'Statutory')]"
        ).match(self.root)[0].get("href")
        stat_response = requests.get(stat_list_url)
        stat_page = lxml.html.fromstring(stat_response.content)
        stat_comm_elements = XPath(
            "//div[@class='grouping-wrapper']//span/a"
        ).match(stat_page)
        all_comm_elements += stat_comm_elements

        for elem in all_comm_elements:
            comm_url = elem.get("href")
            if comm_url[0] == "/":
                comm_url = "https://www.ndlegis.gov" + comm_url
```